src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -131,11 +154,28 @@ protected override void PostPersist(ApplicationEngine engine)
 
             // Distribute GAS for committee
 
-            int index = (int)(engine.Snapshot.PersistingBlock.Index % (uint)ProtocolSettings.Default.CommitteeMembersCount);
+            int m = ProtocolSettings.Default.CommitteeMembersCount;
+            int n = ProtocolSettings.Default.ValidatorsCount;
+            int index = (int)(engine.Snapshot.PersistingBlock.Index % (uint)m);
             var gasPerBlock = GetGasPerBlock(engine.Snapshot);
-            var pubkey = GetCommittee(engine.Snapshot)[index];
+            var pubkey = GetCommitteeFromCache(engine.Snapshot).ElementAt(index);
             var account = Contract.CreateSignatureRedeemScript(pubkey).ToScriptHash();
             GAS.Mint(engine, account, gasPerBlock * CommitteeRewardRatio / 100);
+
+            // Record the cumulative reward of the voters of committee
+
+            var factor = index < n ? 2 : 1;
+            CandidateState candidate = engine.Snapshot.Storages.TryGet(CreateStorageKey(Prefix_Candidate).Add(pubkey))?.GetInteroperable<CandidateState>();
+            if (candidate is null) return;
+            if (candidate.Registered && candidate.Votes > 0)
+            {
+                BigInteger voterSumRewardPerNEO = factor * gasPerBlock * VoterRewardRatio * 100000000L * m / (m + n) / 100 / candidate.Votes; // Zoom in 100000000 times, and the final calculation should be divided 100000000L
```

Comment:
The `voterSumRewardPerNEO` is calculated for each epoch, but `candidate.Votes` can change in every blocks. Will this cause problems?

Reply:
> What if someone changes his votes every block to committee[block.Index % m]?

It's no problem, if in one epoch.

> but candidate.Votes can change in every blocks. Will this cause problems?

We also need to cache `candidate.Votes`  when we cache the committee.